src/DDTrace/Tracer.php

```diff
@@ -430,42 +409,26 @@ private function setPrioritySamplingFromSpan(Span $span)
         }
 
         $prioritySampling = $span->getContext()->getPropagatedPrioritySampling();
-        if (null === $prioritySampling) {
-            $prioritySampling = $this->sampler->getPrioritySampling($span);
+        if (null !== $prioritySampling) {
+            $this->setPrioritySampling($prioritySampling);
         }
-        $this->setPrioritySampling($prioritySampling);
     }
 
     /**
      * @param mixed $prioritySampling
      */
     public function setPrioritySampling($prioritySampling)
     {
-        $this->prioritySampling = $prioritySampling;
-
-        $rootSpan = $this->getSafeRootSpan();
-        if (null === $rootSpan) {
-            return;
-        }
-
-        if ($prioritySampling !== Sampling\PrioritySampling::UNKNOWN) {
-            $rootSpan->setMetric('_sampling_priority_v1', $prioritySampling);
-        } elseif (isset($rootSpan->metrics)) { // official API does not allow removal, so...
-            unset($rootSpan->metrics['_sampling_priority_v1']);
-        }
+        set_priority_sampling($prioritySampling);
+        set_priority_sampling($prioritySampling, true);
     }
 
     /**
      * {@inheritdoc}
      */
     public function getPrioritySampling()
     {
-        // root span is source of truth for priority sampling (if it exists)
-        // @phpstan-ignore-next-line 'Undefined property $metrics' in an isset() check?! Must be a phpstan bug...
-        if (($rootSpan = $this->getSafeRootSpan()) && isset($rootSpan->metrics['_sampling_priority_v1'])) {
-            return $rootSpan->metrics['_sampling_priority_v1'];
-        }
-        return $this->prioritySampling;
+        return get_priority_sampling();
```

Comment:
This PR adds a return type info of `int|null` to `getPrioritySampling` but `DDTrace\get_priority_sampling` is returning `int|false`. This discrepancy needs resolved.

I recommend changing to match `get_priority_sampling` to match `int|null`.

Reply:
Done in 464df10fc19fd78ab0ccb0a5cae63191a5e722f2